src/manifest.c

```diff
@@ -543,7 +543,6 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		ret = 0;
 		goto out;
 	}
-	free(filename);
```

Comment:
With respect, I think this is the wrong fix. Removing this line means a memory leak at about line 561.

I think the correct fix is to add a line so it says

> free(filename);
> filename=NULL;

Of course what we really want is to scrap the string_or_die function, or to add a 

> void free_string(char **s)
> {
>     if (s) {
>        free(*s);
>         *s = NULL;
> }

and use this string_free(&filename) instead of free(filename).

Reply:
Sure, I'll amend this change to reset to NULL instead. Opened #357 regarding a new string_free() function.